src/state.ts

```diff
@@ -137,6 +164,31 @@ export function deleteNamespace(namespace: V1Namespace): void {
   delete state.watchedNamespaces[namespaceName];
 }
 
+function getWorkloadMetadataCacheKey(
+  workloadName: string,
+  namespace: string,
+): string {
+  return `${namespace}/${workloadName}`;
+}
+
+export function getWorkloadMetadata(
```

Comment:
Nit: Could we rename the function to `getCachedWorkloadMetadata`, so we don't have to rename it in `state.ts`?